src/pages/VestingClaim.tsx

```diff
@@ -0,0 +1,191 @@
+import useAddTokenToMetamask from "../hooks/useAddTokenToMetamask"
+
+function VestingClaim(): ReactElement {
+  const { t } = useTranslation()
+  const { account, chainId, library } = useActiveWeb3React()
+  const { addToken, canAdd } = useAddTokenToMetamask({
+    ...SDL_TOKEN,
+    icon: `${window.location.origin}/logo.svg`,
+  })
+
+  const [isValidBene, setIsValidBene] = useState(false)
+  const [claimableVestedAmount, setClaimableVestedAmount] = useState("0.0")
+  const [remainingAmount, setRemainingAmount] = useState("0.0")
+  const [vestingContract, setVestingContract] = useState<Vesting>()
+
+  useEffect(() => {
+    const fetchBeneficiaries = async () => {
+      if (!library || !chainId) return
+      const provider = new ethers.providers.Web3Provider(
+        window.ethereum as never,
+      )
+      const SDLContract = new EthersContract(
+        SDL_TOKEN_ADDRESSES[chainId],
+        SDL_TOKEN_ABI,
+        provider.getSigner(),
+      ) as Sdl
+      const vestingContractDeployedFilter = SDLContract.filters.VestingContractDeployed(
+        null,
+        null,
+      )
+      try {
+        const events = await SDLContract.queryFilter(
+          vestingContractDeployedFilter,
+        )
+        const currentBeneficiary = events.filter(
+          (event) => event.args?.beneficiary === account,
+        )
+        setIsValidBene(currentBeneficiary.length > 0)
+        if (isValidBene) {
+          const vestingContract = new EthersContract(
+            currentBeneficiary[0].args?.vestingContract,
+            INVESTOR_EMPLOYEE_VESTING_CONTRACT_ABI,
+            provider.getSigner(),
+          ) as Vesting
+          setVestingContract(vestingContract)
+          try {
+            const remainingAmount = commify(
+              formatBNToString(
+                await SDLContract.balanceOf(vestingContract.address),
+                18,
+                2,
+              ),
+            )
+            setRemainingAmount(remainingAmount)
+          } catch (err) {
+            console.error(err)
+            notifyCustomError({
+              ...(err as Error),
+              message: "Unable to get total pending amount",
+            })
+          }
+          try {
+            const claimableVestedAmount = commify(
+              formatBNToString(await vestingContract.vestedAmount(), 18, 2),
+            )
+            setClaimableVestedAmount(claimableVestedAmount)
+          } catch (err) {
+            console.error(err)
+            notifyCustomError({
+              ...(err as Error),
+              message: "Unable to get vested amount",
+            })
+          }
+        }
+      } catch (err) {
+        console.error(err)
+        notifyCustomError({
+          ...(err as Error),
+          message: "Unable to query for vesting contracts",
+        })
+      }
+    }
+    void fetchBeneficiaries()
+  }, [account, chainId, library, isValidBene])
+
+  const onClaimClick: () => void = async () => {
```

Comment:
Confirm that this doesn't need to be wrapped in a `useCallback`

Reply:
Wrapping this function in a `useCallback` isn't recommended as explained in https://dmitripavlutin.com/dont-overuse-react-usecallback/